pages/api/waitlist/index.ts

```diff
@@ -19,6 +18,8 @@ import { validateWaitlistRecord } from "@utils/validation/waitlist";
 export default async function handle(req: NextApiRequest, res: NextApiResponse): Promise<void> {
     const session = await getSession({ req });
 
+    console.log(session);
+
     const parentId = session.id as number;
```

Comment:
I think session.id is always a number, you can refer to `types/next-auth.d.ts`
```suggestion
    const parentId = session.id;
```